mlfinlab/backtest_statistics/statistics.py

```diff
@@ -0,0 +1,233 @@
+    # Grouped as min returns by high watermarks
+    high_watermarks = frame.groupby('hwm').min().reset_index()
+    high_watermarks.columns = ['hwm','min']
+    # Time high watermark occured
+    high_watermarks.index = frame['hwm'].drop_duplicates(keep='first').index
+    # Picking ones that had a drawdown after high watermark
+    high_watermarks = high_watermarks[high_watermarks['hwm'] > high_watermarks['min']]
+    if dollars:
+        drawdown = high_watermarks['hwm'] - high_watermarks['min']
+    else:
+        drawdown = 1 - high_watermarks['min'] / high_watermarks['hwm']
+    time_under_water = ((high_watermarks.index[1:] - 
+                         high_watermarks.index[:-1]) / np.timedelta64(1,'Y')).values 
+    time_under_water = pd.Series(time_under_water, index = high_watermarks.index[:-1])
+    return drawdown, time_under_water
+
+
+def sharpe_ratio(returns: pd.Series, cumulative: bool = False, 
+                 entries_per_year:int = 252, risk_free_rate:float = 0) -> float:
+    """
+    Calculates Annualized Sharpe Ratio for pd.Series of returns.
+        
+    :param returns: (pd.Series) returns
+    :param cumulative: (bool) flag if returns are cumulative (no by default)
+    :param entries_per_year: (int) times returns are recorded per year (days by default)
+    :param risk_free_rate: (float) risk-free rate (0 by default)
+    :return: (float) Annualized Sharpe Ratio
+    """  
+    if cumulative:
+        returns = returns.diff() # Converting to simple returns
+    sharpe_ratio = (returns.mean() - risk_free_rate) / returns.std() * (entries_per_year)**(1/2)
+
+    return sharpe_ratio
+
+
+def probabalistic_sharpe_ratio(observed_SR: float, benchmark_SR: float, 
+                               number_of_returns:int , skewness_of_returns:float = 0, 
+                               kurtosis_of_returns: float = 3) -> float:
+    """
+    Calculates the probabilistic Sharpe ratio (PSR) that provides an adjusted estimate of SR, 
+    by removing the inflationary effect caused by short series with skewed and/or 
+    fat-tailed returns.
+    
+    Given a user-defined benchmark Sharpe ratio and an observed Sharpe ratio,
+    PSR estimates the probability that SR ̂is greater than a hypothetical SR. 
+    
+    - It should exceed 0.95, for the standard significance level of 5%. 
+    - It can be computed on absolute or relative returns.
+    
+    :param observed_SR: (float) Sharpe Ratio that is observed
+    :param benchmark_SR: (float) Sharpe Ratio to which observed_SR is tested against
+    :param  number_of_returns: (int) times returns are recorded for observed_SR
+    :param skewness_of_returns: (float) skewness of returns (as Gaussian by default)
+    :param kurtosis_of_returns: (float) kurtosis of returns (as Gaussian by default)
+    :return: (float) Probabalistic Sharpe Ratio
+    """  
+    
+    
+    PSR = ss.norm.cdf(((observed_SR - benchmark_SR) * (number_of_returns - 1)**(1/2)) / \
```

Comment:
Why capital PSR? That is reserved for class names.

Reply:
Will change